controllers/apps/cron_controller.go

```diff
@@ -461,6 +466,85 @@ func (cc *CronController) newEmptyWorkload(apiVersion, kind string) (client.Obje
 	return nil, fmt.Errorf("workload %+v has not implemented client.Object interface", groupVersion)
 }
 
+// Deletes successfully completed jobs beyond given history limits.
+func (cc *CronController) deleteCompletedJobsBeyondThreshold(cron *v1alpha1.Cron) error {
+	historyLimit := cron.Spec.HistoryLimit
+	if historyLimit == nil {
+		return nil
+	}
+
+	completedWorkloads, err := cc.listSuccessfullyCompletedWorkloads(cron)
+	if err != nil {
+		return err
+	}
+
+	if len(completedWorkloads) <= int(*historyLimit) {
+		return nil
+	}
+
+	// Sort completed workloads by creation timestamp.
+	sort.Slice(completedWorkloads, func(i, j int) bool {
+		return completedWorkloads[i].GetCreationTimestamp().Time.Before(completedWorkloads[j].GetCreationTimestamp().Time)
+	})
+
+	// Create a new Scheme object.
+	s := runtime.NewScheme()
+
+	// Register the required Kubernetes API types with the Scheme object.
+	corev1.AddToScheme(s)
+	v1alpha1.AddToScheme(s)
```

Comment:
scheme created here can be a static one and being reused.

Reply:
Fixed it